spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -0,0 +1,451 @@
+            assert folder_path is None, 'Pass either a single folder_path location, or a pair of resfile_path and clufile_path. Ignoring passed folder_path ('+folder_path+').'
+            folder_path, _ = os.path.split(resfile_path)
+        
+        _, SORTING_NAME = os.path.split(folder_path)
+        xml_filepath = "{}/{}.xml".format(folder_path,SORTING_NAME)
+        
+        with open(xml_filepath, 'r') as xml_file:
+            contents = xml_file.read()
+            soup = BeautifulSoup(contents, 'lxml')
+            # Normally, this would be a .xml, but there were strange issues
+            # in the write_recording method that require it to be a .lxml instead
+            # which also requires all capital letters to be removed from the tag names
+        
+        self._sampling_frequency = float(soup.samplingrate.string) # careful not to confuse it with the lfpsamplingsate
+        
+        # Classic functionality reading only a single pair of res and clu files
+        if resfile_path is not None and clufile_path is not None:
+            res = np.loadtxt(resfile_path, dtype=np.int64, usecols=0, ndmin=1)
+            clu = np.loadtxt(clufile_path, dtype=np.int64, usecols=0, ndmin=1)
+            if len(res) > 0:
+                # Extract the number of clusters read as the first line of the clufile then remove it from the clu list
+                n_clu = clu[0]
+                clu = np.delete(clu, 0)
+                unique_ids = np.unique(clu)
+
+                if not unique_ids==np.arange(n_clu+1): # some missing IDs somewhere
+                    if 0 not in unique_ids: # missing unsorted IDs
+                        n_clu += 1
+                    if 1 not in unique_ids: # missing mua IDs
+                        n_clu += 1
+                    # If it is any other ID, then it would be very strange if it were missing...
+
+                # Initialize spike trains and extract times from .res and appropriate clusters from .clu based on user input for ignoring multi-unit activity
+                self._spiketrains = []
+                if keep_mua_units: # default
+                    n_clu -= 1;
+                    self._unit_ids = [x+1 for x in range(n_clu)] # generates list from 1,...,clu[0]-1
+                    for s_id in self._unit_ids:
+                        self._spiketrains.append(res[(clu == s_id).nonzero()])
+                else:
+                    # Ignoring IDs of 0 until get_unsorted_spike_train is implemented into base
+                    # Also ignoring IDs of 1 since user called keep_mua_units=False
+                    n_clu -= 2;
+                    self._unit_ids = [x+1 for x in range(n_clu)] # generates list from 1,...,clu[0]-2
+                    for s_id in self._unit_ids:
+                        self._spiketrains.append(res[(clu == s_id+1).nonzero()]) # only reading cluster IDs 2,...,clu[0]-1
+        elif resfile_path is None and clufile_path is None:
```

Comment:
This should be removed

Reply:
You mean the final 'else' statement at the end of that switch, because the previous assertions mean it would never get called?

Reply:
Disregard. I've adjusted the majority of that logic to be more coherent (hopefully).